docs/docs/optimization.html

```diff
@@ -37,7 +37,9 @@
lass="sidebar-sub-header"><a href="/caliban/docs/optimization.html#using-field-metadata" class="sidebar-link">Using field metadata</a></li><li class="sidebar-sub-header"><a href="/caliban/docs/optimization.html#defer-support-experimental" class="sidebar-link">@defer support (experimental)</a><ul class="sidebar-sub-headers"><li class="sidebar-sub-header"><a href="/caliban/docs/optimization.html#usage" class="sidebar-link">Usage</a></li></ul></li></ul></li><li><a href="/caliban/docs/validation.html" class="sidebar-link">Validation</a></li><li><a href="/caliban/docs/introspection.html" class="sidebar-link">Introspection</a></li><li><a href="/caliban/docs/adapters.html" class="sidebar-link">Http Adapters</a></li><li><a href="/caliban/docs/interop.html" class="sidebar-link">Interop</a></li><li><a href="/caliban/docs/federation.html" class="sidebar-link">Federation</a></li><li><a href="/caliban/docs/relay-connections.html" class="sidebar-link">Relay Connections</a></li><li><a href="/caliban/docs/schema-reporting.html" class="sidebar-link">Schema Reporting</a></li><li><a href="/caliban/docs/server-codegen.html" class="sidebar-link">Server code generation from schema</a></li></ul></section></li><li><section class="sidebar-group collapsable depth-0"><p class="sidebar-heading"><span>Caliban Client</span> <span class="arrow right"></span></p> <!----></section></li><li><section class="sidebar-group collapsable depth-0"><p class="sidebar-heading"><span>Caliban Tools</span> <span class="arrow right"></span></p> <!----></section></li><li><a href="/caliban/docs/examples.html" class="sidebar-link">Examples</a></li></ul> </aside> <main class="page"> <div class="theme-default-content content__default"><h1 id="query-optimization"><a href="#query-optimization" class="header-anchor">#</a> Query optimization</h1> <p>A GraphQL query may request multiple fields that are using the same resolver. It's not a problem if the resolver is a simple value, but it can be less than optimal when the resolver runs an effect (such as reading from a database).</p> <p>We might want to:</p> <ul><li><strong>cache</strong> identical queries (deduplication)</li> <li><strong>batch</strong> queries to the same source</li></ul> <p>This is possible in Caliban using the <a href="https://github.com/zio/zquery" target="_blank" rel="noopener noreferrer"><code>ZQuery</code><span><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" focusable="false" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg> <span class="sr-only">(opens new window)</span></span></a> data type.</p> <p>Additionally, one may want to perform optimizations based on the fields selected by the client.
```

Comment:
This file isn't meant to be modified

Reply:
Ah an Intellij "smart" refactoring did this